### DISCUSSION TITLE
弱网环境下的点对点视频传输

### DISCUSSION THREAD

**Discussion by zhjygit** [Tencent/tquic#73] [General]
想使用自组网电台实现点对点的视频传输，解决弱网环境下的视频传输问题。
是否有相关的测试用例可以参考：比如，前端通过无线摄像头模块采集数据，摄像头借助自组网电台构建的通道，将数据实时传输到后端；具体如何实现呢？能否有可参考的案例用起来呀。

**Comment by wangfuyu** [comment]
> 想使用自组网电台实现点对点的视频传输，解决弱网环境下的视频传输问题。 是否有相关的测试用例可以参考：比如，前端通过无线摄像头模块采集数据，摄像头借助自组网电台构建的通道，将数据实时传输到后端；具体如何实现呢？能否有可参考的案例用起来呀。

你说的是直播全链路（含直播上行上载数据 + 直播下行拉取直播流数据）？ 可否画个简单的网络架构图，以及组件之间期望使用的协议？
根据我们的经验，QUIC在直播上行和下行两个场景，弱网环境下，都是有明显的收益的（秒开率得到提升，卡顿显著减少等）。

**Comment by zhjygit** [comment]
![image](https://github.com/Tencent/tquic/assets/44870751/3952771d-a276-48db-b9d3-66d42f9dd4a0)
前端图传模块，如树莓派和摄像头；
图传模块通过无线网络接入到自组网电台节点1（自组网电台自带无线热点ap模块）；
自组网电台之间透明传输；
后端pc有线网线接入到自组网电台节点2；
弱网环境指的是，点对点自组网传输受到遮挡、通视传输、频率干扰、远距离（5-10公里）等因素；
tquic是否能用在图传模块的数据采集和后端pc的处理方面，使得前端图传采集的视频数据能够稳定地在后端pc上查看或监控。
如果可以的话，测试用例、测试框架能否提供一下，谢谢。

**Reply by wangfuyu** [reply]
QUIC是一种新的通用、安全、多路复用的传输层协议，可用于任何端到端可靠传输场景。
TQUIC是QUIC的一种实现，你们可以将它集成到自己的客户端、服务端程序中，并用在你所列举的场景中，优化节点之间的传输性能。

根据你的描述，实际上您是想要优化『自组网电台节点1』和『自组网电台节点2』之间的网络传输（距离远且非专线，弱网），对吧？这种弱网场景的视频数据传输，QUIC是有较大优势的，如我前面所说，这个场景实际上就类似于弱网直播，根据我们的经验QUIC是有很大优势的。

总的：TQUIC仅是一个QUIC协议实现，需要您按需集成使用，集成可参考示例程序的实现。

**Reply by zhjygit** [reply]
弱网直播，不错呦。参考示例几乎没有参考性呦。毕竟是实时传输啥的。

**Reply by wangfuyu** [reply]
我说的『示例参考』是说框架集成TQUIC库时，可参考示例中对各关键API的使用及上下文关联逻辑哈。
可以这么打个比方，比如你们当前已经有了一套完整的Client<--- xxx(视频协议 flv hls rtmp等) over TCP--->Server系统，可以参考示例程序快速改造成Client<--- xxx(视频协议 flv hls rtmp等) over QUIC --->Server。

QUIC协议相比传统TCP有很多优势，可以大大改善弱网直播体验（成功率、卡顿等），且特别容易落地（QUIC协议栈是在用户态实现，框架改造以及拥塞控制传输优化都很容易）。
<img width="407" alt="image" src="https://github.com/Tencent/tquic/assets/12257285/cc9e0f1e-e690-4e79-88b7-ce2d10dceba7">
